docs/tutorials/booking-system.md

```diff
@@ -0,0 +1,972 @@
+```
+
+Three files should now have been generated inside the `db` folder: `query.sql.go`, `db.go` & `models.go`. These files contain generated Go code and should not be manually edited. We will be adding more queries to `db/query.sql` later and then re-run `sqlc generate` to update the generated Go code.
+
+Now let's create an endpoint that makes use of one of these queries.
+
+🥐 Create `booking/booking.go` with the contents:
+
+```go
+-- booking/booking.go --
+package booking
+
+import (
+	"context"
+	"time"
+
+	"encore.app/booking/db"
+	"github.com/jackc/pgx/v5/pgtype"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"encore.dev/beta/errs"
+	"encore.dev/storage/sqldb"
+)
+
+var (
+	bookingDB = sqldb.NewDatabase("booking", sqldb.DatabaseConfig{
+		Migrations: "./db/migrations",
+	})
+
+	pgxdb = sqldb.Driver[*pgxpool.Pool](bookingDB)
+	query = db.New(pgxdb)
+)
+
+type Booking struct {
+	ID    int64     `json:"id"`
+	Start time.Time `json:"start"`
+	End   time.Time `json:"end"`
+	Email string    `encore:"sensitive"`
+}
+
+type BookParams struct {
+	Start time.Time `json:"start"`
+	Email string    `encore:"sensitive"`
+}
+
+//encore:api public method=POST path=/booking
+func Book(ctx context.Context, p *BookParams) error {
+	eb := errs.B()
+
+	now := time.Now()
+	if p.Start.Before(now) {
+		return eb.Code(errs.InvalidArgument).Msg("start time must be in the future").Err()
+	}
+
+	tx, err := pgxdb.Begin(ctx)
+	if err != nil {
+		return eb.Cause(err).Code(errs.Unavailable).Msg("failed to start transaction").Err()
+	}
+	defer tx.Rollback(context.Background()) // committed explicitly on success
+	
+	_, err = query.InsertBooking(ctx, db.InsertBookingParams{
+		StartTime: pgtype.Timestamp{Time: p.Start, Valid: true},
+		EndTime:   pgtype.Timestamp{Time: p.Start.Add(DefaultBookingDuration), Valid: true},
+		Email:     p.Email,
+	})
+	if err != nil {
+		return eb.Cause(err).Code(errs.Unavailable).Msg("failed to insert booking").Err()
+	}
+
+	if err := tx.Commit(ctx); err != nil {
+		return eb.Cause(err).Code(errs.Unavailable).Msg("failed to commit transaction").Err()
+	}
+	return nil
+}
+```
+
+We are now using the generated type-safe `query.InsertBooking` function to make the database operation.
+
+Notice the `encore:"sensitive"` tag on the `Email` field. This tells Encore to scrub this field so that the data is not viewable in the traces for deployed environments. This is useful for fields that contain [sensitive data](/docs/develop/api-schemas#sensitive-data) such as email addresses, passwords, etc.
+
+🥐 Restart `encore run` to cause the database to be created, and then call the `booking.Book` endpoint:
+
+```shell
+$ curl -X POST 'http://localhost:4000/booking' -d '{"start": "2024-12-11T09:00:00Z", "email": "test@example.com"}'
+```
+
+Congratulations, you have now booked your first appointment!
+
+## 4. Authentication
+
+In order to have an admin dashboard we need to add authentication to our application so that we can have protected endpoints. 
```

Comment:
```suggestion
To provide an admin dashboard for our booking system, we need to add authentication to our application so that we can have protected endpoints.

Keep in mind, in this tutorial we'll only include a very basic implementation.
```